bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/ThingHandlerService.java

```diff
@@ -13,11 +13,25 @@
 package org.eclipse.smarthome.core.thing.binding;
 
 /**
+ * The {@link ThingHandler#getServices()} function allows you to return a list of classes,
+ * implementing this interface, that will be started as OSGi services. 
+ * <p>
+ * Use this for example to start thing specific discovery services (extending {@link AbstractDiscoveryService}).
+ * Example:
+ * <pre>
+ * @NonNullByDefault
+ * public class MyThingDiscovery extends AbstractDiscoveryService implements ThingHandlerService {
```

Comment:
We have to slightly change this as the important thing is to implement the interface which has to be registered as service:

```suggestion
 * public class MyThingDiscovery extends AbstractDiscoveryService implements DiscoveryService, ThingHandlerService {
```